distributed/scheduler.py

```diff
@@ -3596,6 +3596,84 @@ def transition_no_worker_waiting(self, key):
                 pdb.set_trace()
             raise
 
+    def decide_worker_initial(self, ts: TaskState) -> WorkerState:
+        r"""
+        Decide the worker to assign a task to.
+
+        Parameters
+        ----------
+        ts : TaskState
+            This is a ready task that has no dependencies.
+
+        Returns
+        -------
+        WorkerState
+
+        Notes
+        -----
+        This prioritizes scheduling a task on a worker executing tasks that we're
+        a sibling dependency with.
+
+        Consider the following task graph (read top to bottom)
+
+        ::
+
+            a-1  a-2  a-3  a-4
+             \   /     \   /
+              b-1       b-2
```

Comment:
Note that all ascii art diagrams in the codebase so far have computation going from bottom to top.  This is also the way that visualize works.